tls-client-with-opaque-keys/main.cpp

```diff
@@ -0,0 +1,446 @@
+/*
```

Comment:
This should be `tls-client-with-authentication` or `tls-client-with-client-authentication`. In the medium term it'll support reading a private key from a file.